### DISCUSSION TITLE
Technologies Discussion

### DISCUSSION THREAD

**Discussion by Gaby-Concepcion** [uprm-inso4116-2024-2025-s1/semester-project-multiplayer-dominoes#1] [General]
**DISCUSSION ON UTILIZED TECHNOLOGIES**

A discussion post on what frameworks or other resources will be used during the project.
Technologies proposed to be used in the Project proposal:
-     Frontend: React.js, WebSocket for real-time communication
-     Backend: Node.js, Express.js, MongoDB
-     Cloud Services: AWS or Google Cloud for hosting and database management
-     Tools: GitHub for version control, Figma for design prototyping

Given these as a guideline or concept, I was hoping to use this discussion forum so that the teams can discuss what best fits their needs.
Teams as they stand right this moment (Aug. 27, 9:17:50 pm):
- **GameState/AI Team**
- **Sign In/Log In**
- **Graphic Design**

**Comment by PinkSylvie** [comment]
I see no issues with the proposed technologies

**Comment by fernandopizarro1** [comment]
I think a good cloud service could be firebase, since its really simple to use and it fits our needs. Also, I think React.js and node.js are perfect for the frontend and backend

**Comment by print-Eruki** [comment]
We need a WebSocket server to enable real-time communication between players. The game's rules and calculations should be handled on the server-side to ensure consistency. This means that client-side actions will be sent to the server for processing, and the resulting updates are going to send events to the clients for handling

Are we looking into free hosting services or local only?
